src/db/index.ts

```diff
@@ -76,26 +74,16 @@ export async function findHostIssues (knex: Knex, host_id: string) {
     .where(RemediationIssueSystems.system_id, EQ, host_id);
 }
 
-function createUpdateQuery (host_id: string, issue_id: string, update: string) {
-    const updateQuery = `UPDATE remediation_issue_systems SET resolved = ${update} ` +
-    `FROM remediation_issues ` +
-    `WHERE remediation_issues.id = remediation_issue_systems.remediation_issue_id ` +
-    `AND issue_id = '${issue_id}' ` +
-    `AND remediation_issue_systems.system_id = '${host_id}'`;
-
-    return updateQuery;
-}
-
-export async function updateToUnresolved (knex: Knex, host_id: string, issue_id: string) {
-    const QUERY = createUpdateQuery(host_id, issue_id, UNRESOLVED);
-
-    return knex.raw(QUERY);
-}
-
-export async function updateToResolved (knex: Knex, host_id: string, issue_id: string) {
-    const QUERY = createUpdateQuery(host_id, issue_id, RESOLVED);
-
-    return knex.raw(QUERY);
+export async function updateIssues (knex: Knex, host_id: string, issues: string[], prefix: string) {
+    return knex.raw(
+        `UPDATE remediation_issue_systems SET resolved = remediation_issues.issue_id ` +
+        'NOT IN (' + issues.map(() => '?').join(',') + ') ' +
```

Comment:
it would be cleaner for the template to be static i.e. not change based on the issues parameter, e.g:
```
NOT IN (?)
```
and then pass
```
issues.join(',')
```
as the parameter.

Reply:
Also, any time the function is invoked all the strings need to be concatenated first. It would be better for it to be a single string or a constant that is prepared upfront.